src/bokeh/models/tools.py

```diff
@@ -396,8 +401,12 @@ def __init__(self, *args, **kwargs) -> None:
     If True, hides toolbar when cursor is not in canvas.
     """)
 
-    tools = List(Either(Instance(Tool), Instance(ToolProxy)), help="""
-    A list of tools to add to the plot.
+    group = Bool(default=True, help="""
+    Whether to group common tools by type.
+    """)
+
+    group_types = List(Enum(ToolName), default=["hover", "tap"], help="""
```

Comment:
Maybe only add  `hover` for now? Or does the issue ask for taptool too?

Reply:
The issue asked for tool grouping in general and specifically for hover. In any case, I need to resolve issue #3804 first.